From 897ada2a7128ba0180d03dfc7c729fd765cdb13e Mon Sep 17 00:00:00 2001
From: Josua Mayer <josua@solid-run.com>
Date: Sun, 10 Mar 2024 09:20:45 +0100
Subject: [PATCH] (HACK) drivers: renesas: micro-delay: count from 10 million

calculation of accurate loop count based on clock frequency locked up
the system. Replace by simple counter from 10 million.

Signed-off-by: Josua Mayer <josua@solid-run.com>
---
 drivers/renesas/common/delay/micro_delay.c | 10 +---------
 1 file changed, 1 insertion(+), 9 deletions(-)

diff --git a/drivers/renesas/common/delay/micro_delay.c b/drivers/renesas/common/delay/micro_delay.c
index a5e2a692..a6739b24 100644
--- a/drivers/renesas/common/delay/micro_delay.c
+++ b/drivers/renesas/common/delay/micro_delay.c
@@ -17,15 +17,7 @@ void
 #endif
 	rcar_micro_delay(uint64_t micro_sec)
 {
-	uint64_t freq;
 	uint64_t base_count;
-	uint64_t get_count;
-	uint64_t wait_time = 0U;
 
-	freq = read_cntfrq_el0();
-	base_count = read_cntpct_el0();
-	while (micro_sec > wait_time) {
-		get_count = read_cntpct_el0();
-		wait_time = ((get_count - base_count) * RCAR_CONV_MICROSEC) / freq;
-	}
+	for (base_count = 10000000; base_count > 1; base_count--);
 }
-- 
2.35.3

